src/local/room_position.rs

```diff
@@ -0,0 +1,375 @@
+            x_coord: self.room_x(),
+            y_coord: self.room_y(),
+        }
+    }
+
+    #[inline]
+    pub fn set_x(&mut self, x: u32) {
+        assert!(x < 50, "out of bounds x: {}", x);
+        self.packed = (self.packed & !(0xFF << 8)) | (x << 8);
+    }
+
+    #[inline]
+    pub fn set_y(&mut self, y: u32) {
+        assert!(y < 50, "out of bounds y: {}", y);
+        self.packed = (self.packed & !0xFF) | y;
+    }
+
+    #[inline]
+    pub fn set_room_name(&mut self, room_name: LocalRoomName) {
+        assert!(
+            (-128..=127).contains(&room_name.x_coord),
+            "out of bounds room_x: {}",
+            room_name.x_coord,
+        );
+        assert!(
+            (-128..=127).contains(&room_name.y_coord),
+            "out of bounds room_y: {}",
+            room_name.y_coord,
+        );
+        let room_x = (room_name.x_coord + HALF_WORLD_SIZE) as u32;
+        let room_y = (room_name.y_coord + HALF_WORLD_SIZE) as u32;
+
+        self.packed = (self.packed & 0xFFFF) | (room_x << 24) | (room_y << 16);
+    }
+
+    #[inline]
+    pub fn with_x(mut self, x: u32) -> Self {
+        self.set_x(x);
+        self
+    }
+
+    #[inline]
+    pub fn with_y(mut self, y: u32) -> Self {
+        self.set_y(y);
+        self
+    }
+
+    #[inline]
+    pub fn with_room_name(mut self, room_name: LocalRoomName) -> Self {
+        self.set_room_name(room_name);
+        self
+    }
+}
+
+mod stdweb {
+    use stdweb::{Reference, Value};
+
+    use crate::{
+        macros::*,
+        objects::{HasPosition, RoomPosition},
+        traits::{TryFrom, TryInto},
+    };
+
+    use super::LocalRoomPosition;
+
+    impl LocalRoomPosition {
+        pub fn remote(self) -> RoomPosition {
+            js_unwrap_ref!(pos_from_packed(@{self.packed_repr()}))
+        }
+    }
+
+    impl TryFrom<Value> for LocalRoomPosition {
+        type Error = <Value as TryInto<String>>::Error;
+
+        fn try_from(v: Value) -> Result<LocalRoomPosition, Self::Error> {
+            if let Value::Number(v) = v {
+                let packed: i32 = v.try_into()?;
+                return Ok(LocalRoomPosition::from_packed(packed));
+            }
+
+            let value = js! { return @{v}.__packedPos };
+            match value {
+                Value::Undefined => {
+                    let x = js! {v.x}.try_into()?;
+                    let y = js! {v.y}.try_into()?;
+                    let room_name = js! {v.roomName}.try_into()?;
+                    Ok(Self::new(x, y, room_name))
+                }
+                other => Ok(Self::from_packed(other.try_into()?)),
+            }
+        }
+    }
+
+    impl crate::traits::FromExpectedType<Reference> for LocalRoomPosition {
+        fn from_expected_type(reference: Reference) -> Result<Self, crate::ConversionError> {
+            Self::try_from(Value::Reference(reference))
+        }
+    }
+
+    impl HasPosition for LocalRoomPosition {
+        fn pos(&self) -> RoomPosition {
```

Comment:
(For later) maybe consider having `pos` return a `LocalRoomPosition`. Then, it may make more sense to have `RoomPosition` be `RemoteRoomPosition`.

Reply:
Planning on having this change in the next round, once we get rid of the "remote" room position.